zaza/openstack/charm_tests/aodh/tests.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python3
+
+# Copyright 2019 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Encapsulate Aodh testing."""
+
+import logging
+
+import zaza.model
+import zaza.openstack.charm_tests.test_utils as test_utils
+import zaza.openstack.utilities.openstack as openstack_utils
+import zaza.openstack.configure.telemetry as telemetry_utils
+
+
+class AodhTest(test_utils.OpenStackBaseTest):
+    """Encapsulate Aodh tests."""
+
+    RESOURCE_PREFIX = 'zaza-aodhtests'
+
+    @classmethod
+    def setUpClass(cls):
+        """Run class setup for running tests."""
+        super(AodhTest, cls).setUpClass()
+        cls.xenial_ocata = openstack_utils.get_os_release('xenial_ocata')
+        cls.xenial_newton = openstack_utils.get_os_release('xenial_newton')
+        cls.bionic_stein = openstack_utils.get_os_release('bionic_stein')
+        cls.release = openstack_utils.get_os_release()
+        cls.keystone_session = openstack_utils.get_overcloud_keystone_session()
+        cls.model_name = zaza.model.get_juju_model()
+        cls.aodh_client = openstack_utils.get_aodh_session_client(
+            cls.keystone_session)
+
+    @classmethod
+    def tearDown(cls):
+        """Remove test resources."""
+        logging.info('Running teardown')
+        cache_wait = False
+        for alarm in cls.aodh_client.alarm.list():
+            if alarm['name'].startswith(cls.RESOURCE_PREFIX):
+                cache_wait = True
+                logging.info('Removing Alarm {}'.format(alarm['name']))
+                telemetry_utils.delete_alarm(
+                    cls.aodh_client,
+                    alarm['name'],
+                    cache_wait=False)
+        if cache_wait:
+            logging.info('Waiting for alarm cache to clear')
+            telemetry_utils.alarm_cache_wait()
+
+    @property
+    def services(self):
```

Comment:
Is there a way to test the functionality these services provide rather than setting us up for maintenance of this ever evolving set of process names?

Reply:
I do agree that this in not very black-box. But the pause and resume test in particular needs to ascertain that service are not running when the unit is paused. We could pause a unit an then test that the api wasn't responding and/or an alarm fails to trigger but it feels quite brittle.

Reply:
:+1: